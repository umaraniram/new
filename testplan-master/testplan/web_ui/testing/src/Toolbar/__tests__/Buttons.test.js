/**
 * Unit tests for the Buttons module
 */
import React from "react";
import {shallow} from "enzyme";
import {StyleSheetTestUtils} from "aphrodite";

import {TimeButton} from "../Buttons";

describe("TimeButton", () => {
  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterEach(() => {
    // Resume style injection once test is finished.
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it("Renders a clickable button", () => {
    const updateTimeDisplayCbk = jest.fn();
    const button = shallow(
      <TimeButton updateTimeDisplayCbk={updateTimeDisplayCbk} />
    );
    expect(button).toMatchSnapshot();
    button.find({title: "Display time information"}).simulate("click");
    expect(updateTimeDisplayCbk.mock.calls.length).toBe(1);
  });
});
