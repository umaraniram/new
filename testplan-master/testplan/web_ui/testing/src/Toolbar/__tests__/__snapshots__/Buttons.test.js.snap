// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeButton Renders a clickable button 1`] = `
<NavItem
  tag="li"
>
  <div
    className="buttonsBar_6zsntc"
  >
    <FontAwesomeIcon
      border={false}
      className="toolbarButton_1d6e8oq"
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f017",
            "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z",
          ],
          "iconName": "clock",
          "prefix": "fas",
        }
      }
      inverse={false}
      key="toolbar-time"
      listItem={false}
      mask={null}
      onClick={[Function]}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title="Display time information"
      transform={null}
    />
  </div>
</NavItem>
`;
